libraries/AP_MSP/AP_MSP.cpp

```diff
@@ -86,7 +86,7 @@ void AP_MSP::init()
     for (uint8_t protocol_instance=0; protocol_instance<MSP_MAX_INSTANCES; protocol_instance++) {
```

Comment:
if we don't use the protocol_instance variable, maybe using 

```suggestion
    while (_msp_status.backend_count < MSP_MAX_INSTANCES) {
```
would be better ?

Reply:
@khancyr I refactored and removed backend_instance, now should be ok. It loops thru all instances per each serial protocol starting from instance 0 up to MAX  instances accounting for already found backends